source/StartConditions.cpp

```diff
@@ -300,81 +247,124 @@ const Conversation &StartConditions::GetConversation() const
 
 
 
-const Sprite *StartConditions::GetThumbnail()
+const Sprite *StartConditions::GetThumbnail() const noexcept
 {
-	return infoByState[state].thumbnail;
+	// The state we ask for should always be present in the map, but just in case it isn't,
+	// make sure we're not accessing out of bound information.
+	auto it = infoByState.find(state);
+	return it == infoByState.end() ? nullptr : it->second.thumbnail;
 }
 
 
 
-const std::string &StartConditions::GetDisplayName()
+const string &StartConditions::GetDisplayName() const noexcept
 {
-	return infoByState[state].name;
+	auto it = infoByState.find(state);
+	static const string ILLEGAL = "(ILLEGAL STATE)";
```

Comment:
illegal state? do we use this to contain information? shouldnt it be in the unlocked state since that one would have everything?
I never see it being set; this seems like it wont work to me?

Reply:
I'd rather return something that explicitly states that we've somehow messed up somewhere and need to look into what happened. "???" could easily be an actual string that someone wants to use.

Reply:
you're right this should never happen anyway, at least if it does we'll get a bug report